src/api.v2/model/Experiment.js

```diff
@@ -144,6 +144,24 @@ class Experiment extends BasicModel {
     }
   }
 
+  async getProcessingConfig(experimentId) {
+    const result = await this.find({ id: experimentId });
+    if (_.isEmpty(result)) {
+      throw new NotFoundError('Experiment not found');
+    }
+
+    return result[0].processingConfig;
+  }
+
+  async updateProcessingConfig(experimentId, body) {
+    const { name: stepName, body: change } = body[0];
+
+    const processingConfig = await this.getProcessingConfig(experimentId);
+    processingConfig[stepName] = change;
+
+    await this.update(experimentId, { processing_config: processingConfig });
```

Comment:
I had a conversation with Martin about doing JSONB update in-place, instead of querying, updating and saving it again. `processing_config` is also a `jsonb` field, so you should be able to update it [in-place](https://stackoverflow.com/questions/48888792/update-or-create-nested-jsonb-value-using-single-update-command). I'm not sure if we're finally going ahead with that approach. Have you talked to @cosa65 about this?

Reply:
Discussed with Martin. For now I think we should avoid making big changes to the UI, we could create PATCH requests in the future for this. Maybe I should create a ticket to do this?

Reply:
Oh, I think I might have misunderstood the original question, my response was more along the lines of not making the UI more complex.

I see what this is doing is actually already a `PATCH` (and what the UI is doing is also a `PATCH` even though the endpoint says `PUT`) as in it is only changing one property within the `processingConfig`, not setting a whole new `procesingConfig`.

In this case I think I agree with agi, trying to do something like what we did with dynamo (only one query to update a property) might be better.

I'm so sorry for the confusion!! I hadn't had time to actually take a look at the PR and check what we had discussed, I should have checked the code first